lib/src/iterable_matchers.dart

```diff
@@ -206,6 +215,26 @@ class _UnorderedMatches extends Matcher {
   Description describeMismatch(item, Description mismatchDescription,
           Map matchState, bool verbose) =>
       mismatchDescription.add(_test(item));
+
+  /// Returns [true] if the value at [valueIndex] can be paired with some
+  /// unmatched matcher.
+  ///
+  /// Recursively looks for new pairings whenever there is a conflict. [seen]
+  /// tracks the matchers that have already been consumed within this search.
+  bool _findPairing(List<List<bool>> adjacency, int valueIndex, Set<int> seen,
+      List<int> matched) {
+    for (int i = 0; i < matched.length; i++) {
+      if (adjacency[valueIndex][i] && !seen.contains(i)) {
```

Comment:
Note that my suggestion above (a list of ints instead of bools for `adjacency`) would make this loop easier and more efficient also (just iterate `adjacency[valueIndex]`).

Reply:
Yeah I like they way that looks. Also lets me pull the `seen.contains` check out of the loop into an `Iterable.where` which I think helps.